modules/core/src/main/java/org/apache/ignite/internal/managers/discovery/GridDiscoveryManager.java

```diff
@@ -3082,7 +3115,7 @@ else if (log.isDebugEnabled())
                             evt.discoCache = discoCache;
                         }
 
-                        ctx.event().record(customEvt, evt.discoCache);
+                        withContextIfNeed(evt.secSubjId, ctx.security(), () -> ctx.event().record(customEvt, evt.discoCache));
```

Comment:
We should change the security context for the whole `DiscoveryWorker`

Reply:
ok, done